### DISCUSSION TITLE
The demo code has a lot of duplications

### DISCUSSION THREAD

**Discussion by the-gigi** [sashirestela/simple-openai#104] [Ideas]
The OpenAIAzure and OpenAIAnyscale demos are pretty much copy-pasta of the original OpenAI.

There is a lot of room for refactoring, as well as making it explicit, which capabilities are supported by each provider.

For example, at the moment Azure OpenAI doesn't support the streaming flavor and Anyscale doesn't support chat+vision.

**Comment by sashirestela** [comment]
@the-gigi SonarCloud's quality metrics have worsened after merging your PR https://github.com/sashirestela/simple-openai/pull/92. Demo code is not a priority right now. The code to support other vendors needs an overall refactor. Please, don't make more contributions until more news.

**Comment by sashirestela** [comment]
@the-gigi With the last refactor for handling multiple providers the demo code is fully reusable
